bigtable/bigtable-proxy/README.md

```diff
@@ -0,0 +1,88 @@
+all processes on the VM that it is running on. It's listening address is hardcoded to `localhost`.
+The proxy will use [Application Default Credentials](https://cloud.google.com/docs/authentication/application-default-credentials)
+for all outbound RPCs.
+
+The proxy will accept local unencrypted connections from Bigtable clients, and:
+- attach credentials
+- export metrics
+- send the RPC over an encrypted channel pool to Bigtable service
+
+## Features
+
+* Metrics - The proxy will track RPC metrics and export them to Google Cloud Monitoring
+* Multi tenant - The proxy can be used to connect to many different Bigtable instances
+* Credential handling - The proxy has its own set of credentials. It will ignore any inbound
+  credentials from the client
+* Channel pooling - The proxy will maintain and autosize the outbound channel pool to properly
+  load balance RPCs.
+
+## Metrics
+
+The proxy is instrumented with Opentelemtry and will export those metrics to Google Cloud Monitoring
+in a project your choosing. The metrics will be published under the namespace
+`workload.googleapis.com`. Available metrics:
+
+* `bigtableproxy.server.call.started` The total number of RPCs started, including those that have 
+    not completed.
+* `bigtableproxy.client.call.credential.duration` Latency of getting credentials
+* `bigtableproxy.client.call.queue.duration` Duration of how long the outbound side of the proxy had
+  the RPC queued
+* `bigtableproxy.client.call.sent_total_message_size` Total bytes sent per call to Bigtable service
+  (excluding metadata, grpc and transport framing bytes
+* `bigtableproxy.client.call.rcvd_total_message_size` Total bytes received per call from Bigtable 
+  service (excluding metadata, grpc and transport framing bytes)
+* `bigtableproxy.client.gfe.duration` Latency as measured by Google load balancer from the time it 
+  received the first byte of the request until it received the first byte of the response from the
+  Cloud Bigtable service.
+* `bigtableproxy.client.gfe.duration_missing.count` Count of calls missing gfe response headers
+* `bigtableproxy.client.call.duration` Total duration of how long the outbound call took
+* `bigtableproxy.client.channel.count` Number of open channels
+* `bigtableproxy.client.call.max_outstanding_count` Maximum number of concurrent RPCs in a single
+  minute window
+
+## Requirements
+
+* JVM >= 11
+* Ensure that the service account includes the IAM roles:
+  * `Monitoring Metric Writer`
+  * `Bigtable User`
+* Ensure that the metrics project has `Stackdriver Monitoring API` enabled
+
+## Expected usage
+
+```sh
+# Build the binary
+mvn package
+
+# use the binary
+unzip target/bigtable-proxy-0.0.1-SNAPSHOT-bin.zip
+cd bigtable-proxy-0.0.1-SNAPSHOT
+./bigtable-proxy.sh \
+  --listen-port=1234 \
+  --metrics-project-id=SOME_GCP_PROJECT
+ 
+export BIGTABLE_EMULATOR_HOST=1234
+path/to/application/with/bigtable/client
```

Comment:
can these be one line?

export BIGTABLE_EMULATOR_HOST=1234 path/to/application/with/bigtable/client

Reply:
they could be, but I prefer to make it explicit